### ISSUE TITLE
Negative weight is accepted

### ISSUE THREAD

**Issue by graywolf** [dgryski/go-ketama#2]
It is possible to create a bucket with negative weight leading to this module not mapping anything:

```
package main

import (
        "fmt"

        "github.com/dgryski/go-ketama"
)

func main() {
        var buckets []ketama.Bucket

        buckets = append(buckets, ketama.Bucket{"127.0.0.1", -300})
        buckets = append(buckets, ketama.Bucket{"127.0.0.1:11212", 200})
        buckets = append(buckets, ketama.Bucket{"127.0.0.1:11213", 100})

        fmt.Println(buckets)

        c, _ := ketama.New(buckets)

        tests := []string{
                "thing1",
                "thing2",
                "thing3",
                "thing4",
                "thing5",
                "thing6",
                "thing7",
                "thing8",
                "thing9",
                "thing0",
                "thinga",
                "thingb",
                "thingc",
                "thingd",
        }
        for _, test := range tests {
                fmt.Printf("%q -> %q\n", test, c.Hash(test))
        }
}
```

when executed:

```
+   $ go run git.sr.ht/~graywolf/gomemcache/foo
[{127.0.0.1 -300} {127.0.0.1:11212 200} {127.0.0.1:11213 100}]
"thing1" -> ""
"thing2" -> ""
"thing3" -> ""
"thing4" -> ""
"thing5" -> ""
"thing6" -> ""
"thing7" -> ""
"thing8" -> ""
"thing9" -> ""
"thing0" -> ""
"thinga" -> ""
"thingb" -> ""
"thingc" -> ""
"thingd" -> ""
```

Since weight <= 0 does not really make sense, I would propose `New()` to return an error if such a weight is provided. If you agree I can prepare a patch.

**Comment by dgryski** [comment]
That seems reasonable.  I'll wait for your PR.

**Comment by graywolf** [comment]
I've opened #4 , let me know what you think.

**Comment by dgryski** [comment]
Fixed in #4
